vortex-datafusion/src/plans.rs

```diff
@@ -121,17 +124,21 @@ impl ExecutionPlan for RowSelectorExec {
         partition: usize,
         _context: Arc<TaskContext>,
     ) -> DFResult<SendableRecordBatchStream> {
-        assert_eq!(
-            partition, 0,
-            "single partitioning only supported by RowSelectorExec"
-        );
+        if partition != 0 {
+            return Err(vortex_err!(
+                "Single partitioning only supported by RowSelectorExec, got partition {}",
+                partition
+            )
+            .into());
```

Comment:
vortex_bail!().into?

Reply:
doesn't work because that macro has `return` inside it, order of ops is wrong